source/uwp/Visualizer/ViewModel/GenericDocumentViewModel.cs

```diff
@@ -117,15 +120,58 @@ protected async Task LoadFromFileAsync(IStorageFile file, string token, bool ass
             }
         }
 
+        private async Task CreateMessageDialog(string text)
+        {
+            var dontWait = new MessageDialog(text);
+            dontWait.CancelCommandIndex = 0;
+            await dontWait.ShowAsync();
+        }
+
+        private async Task SaveNewFile()
+        {
+            var savePicker = new FileSavePicker();
+            savePicker.SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
+            savePicker.FileTypeChoices.Add("Plain Text", new List<string>() { ".json" });
+            savePicker.SuggestedFileName = "NewAdaptiveCard.json";
+
+            File = await savePicker.PickSaveFileAsync();
+            if (File != null)
+            {
+                CachedFileManager.DeferUpdates(File);
+                await FileIO.WriteTextAsync(File, File.Name);
+                FileUpdateStatus status = await CachedFileManager.CompleteUpdatesAsync(File);
+                if (status == FileUpdateStatus.Complete)
+                {
+                    this.Name = File.Name;
+                    await CreateMessageDialog("File " + File.Name + " was saved");
+                }
+                else
+                {
+                    await CreateMessageDialog("File " + File.Name + " couldn't be saved");
+                }
+            }
+            else
+            {
+                await CreateMessageDialog("Save new file was cancelled");
```

Comment:
>CreateMessageDialog [](start = 22, length = 19)

Do we need another dialog to pop up in this case? Generally when i click cancel i expect to go back to the app, not see another dialog. Similarly for the success case, i'm not sure we need a dialog there either. Maybe just for the error case? What do you think?

Reply:
I can agree with the cancel dialog not showing. On the case of saving success and fail I wonder if not getting feedback on what's happening may be the best scenario. I could remove everything and if we receive a new bug I can re add them

---
In reply to: [389912842](https://github.com/microsoft/AdaptiveCards/pull/3825#discussion_r389912842) [](ancestors = 389912842)